RoddyCore/src/de/dkfz/roddy/core/ProjectLoader.groovy

```diff
@@ -457,8 +459,26 @@ class ProjectLoader {
                 sb << "  " << projectID << "@" << aID << Constants.ENV_LINESEPARATOR;
             }
             throw new ProjectLoaderException(sb.toString());
-        } else if (analysis.getRuntimeService() == null) {
+        }
+
+        if (analysis.getRuntimeService() == null) {
             throw new ProjectLoaderException("There is no runtime service class set for the selected analysis. This has to be set in either the project configuration or the analysis configuration.");
         }
+
+        // Earliest check for valid input and output directories. If they are not accessible or writeable.
+        // Start with the input directory
+        for (File _dir = analysis.getInputBaseDirectory(); _dir; _dir = _dir.parentFile) {
+            if (!FileSystemAccessProvider.instance.isReadable(_dir))
+                throw new ProjectLoaderException("The input directory was not readable at path ${_dir}.")
+        }
+
+        for (File _dir = analysis.getOutputBaseDirectory(); _dir; _dir = _dir.parentFile) {
+            if (!FileSystemAccessProvider.instance.isReadable(_dir))
```

Comment:
The output directory does not need to be writable, right? Some files may go into existing (writable) subdirectories, right?

Reply:
It must be because the dataset folders will be created here, also the execution cache dir and the execution list. If they were already create ok, we could skip it. But then it is in the context and I think it might already be checked there.

Reply:
So, you should add a test for writability at the terminal output directory then.

BTW: The error message is wrong (says input directory).